cmd/monaco/integrationtest/v2/settings_integration_test.go

```diff
@@ -68,5 +74,57 @@ func TestIntegrationValidationSettings(t *testing.T) {
 	cmd.SetArgs([]string{"deploy", "--verbose", "--dry-run", manifest})
 	err := cmd.Execute()
 
-	assert.NilError(t, err)
+	assert.NoError(t, err)
+}
+
+// TestOldExternalIDGetsUpdated tests whether a settings object with an "old" external ID that was
+// generated using only "schemaID" and "configID" gets recognized and updated to have the "new" external ID
+// that is composed of "projectName", "schemaID" and "configID"
+func TestOldExternalIDGetsUpdated(t *testing.T) {
+	fs := testutils.CreateTestFileSystem()
+	var manifestPath = "test-resources/integration-settings-old-new-external-id/manifest.yaml"
+	loadedManifest := integrationtest.LoadManifest(t, fs, manifestPath, "")
+	projects := integrationtest.LoadProjects(t, fs, manifestPath, loadedManifest)
+	sortedConfigs, _ := topologysort.GetSortedConfigsForEnvironments(projects, []string{"platform_env"})
+	environment := loadedManifest.Environments["platform_env"]
+	configToDeploy := sortedConfigs["platform_env"][0]
+
+	t.Cleanup(func() {
+		integrationtest.CleanupIntegrationTest(t, fs, manifestPath, loadedManifest, "")
+	})
+
+	// first deploy with external id generate that does not consider the project name
+	c, _ := dynatrace.CreateDTClient(environment.URL.Value, environment.Auth, false, dtclient.WithExternalIDGenerator(func(input coordinate.Coordinate) string {
+		input.Project = ""
+		return idutils.GenerateExternalID(input)
+	}))
+
+	_, err := c.UpsertSettings(dtclient.SettingsObject{
+		Coordinate:     configToDeploy.Coordinate,
+		SchemaId:       configToDeploy.Type.(v2.SettingsType).SchemaId,
+		SchemaVersion:  configToDeploy.Type.(v2.SettingsType).SchemaVersion,
+		Scope:          "environment",
+		Content:        []byte(configToDeploy.Template.Content()),
+		OriginObjectId: configToDeploy.OriginObjectId,
+	})
+	assert.NoError(t, err)
+
+	// then deploy same config with external id generator that consider the project name as well
+	c, _ = dynatrace.CreateDTClient(environment.URL.Value, environment.Auth, false)
+	_, err = c.UpsertSettings(dtclient.SettingsObject{
+		Coordinate:     configToDeploy.Coordinate,
+		SchemaId:       configToDeploy.Type.(v2.SettingsType).SchemaId,
+		SchemaVersion:  configToDeploy.Type.(v2.SettingsType).SchemaVersion,
+		Scope:          "environment",
+		Content:        []byte(configToDeploy.Template.Content()),
+		OriginObjectId: configToDeploy.OriginObjectId,
+	})
+	assert.NoError(t, err)
+
+	settings, _ := c.ListSettings("builtin:anomaly-detection.metric-events", dtclient.ListSettingsOptions{
+		DiscardValue: true,
+	})
+
+	// assert that ther is only 1 settings object
+	assert.Len(t, settings, 1)
```

Comment:
This assert seems like it will be pretty flaky - if any other test has currently created a metric-event, or some cleanup failed, there might be more than one. 

You could pass a filter on the expected externalId to ListSettings and check that the legacy ID is not found, but the new one is